Adds "monster tamer" drink, plus some chimera tweaks.

Screemonster opened the pull request:
- Adds "monster tamer", a drink made by mixing animal protein and whiskey. Combination of good ol' meat with a neural depressant calms them down from feralness faster (provided they're not still in the state that MADE them feral)

- Renamed "begin reconstitute form" into just "reconstitute form".

- Moved their revive into its own proc so it can be called without the verb.

- Added a chemical reaction (5u monster tamer, 5u clonexadone) that will do the above. Will give feedback as to whether it was successful or not, ie. whether they have nutriment in them, aren't reviving already. Has no effect on live ones. Mainly exists so the chimera in deadchat don't have to keep returning to their body and mashing the button to see if they've been injected with enough nutriment yet.

- Made them NO_INFECT, removed Purge Impurities. (no slyly removing toxloss with a verb that's meant to reflect their strong immune systems. They have actually strong immune systems now)

- Moved handle_feral calls under a spawn(0) because the first thing that happens in that proc after the "am I handling hallucinations already" check is a sleep() and that doesn't seem like a good thing to have happening in a critter's life() ticks. Removed redundant if(!handling_hal) checks because the first thing in the proc they call is that exact check.

Screemonster commented:
Also just spotted and fixed a bug that popped up in testing and for the life of me I couldn't reproduce deliberately - namely that really minuscule (but still positive) values for feral were halting life, as handle_feral() had a sleep(rand(200,500)/(feral/10)) statement in it.

So a feral value of 4E-6 (which is what it had when it happened) was making it sleep _for goddamn ever_.

Have made handle_feral outright reject feral values under 10 so the maximum delay is 500.

Arokha commented on code/modules/mob/living/carbon/human/species/station/station_special_abilities_vr.dm:
```diff
@@ -209,7 +158,7 @@
 
-	if(client && feral) // largely a copy of handle_hallucinations() without the fake attackers. Unlike hallucinations, only fires once - if they're still feral they'll get hit again anyway.
+	if(client && feral >= 10) // largely a copy of handle_hallucinations() without the fake attackers. Unlike hallucinations, only fires once - if they're still feral they'll get hit again anyway.
 		sleep(rand(200,500)/(feral/10))
```
Do not sleep in Life() ever. It sleeps the entire mob controller. When you sleep, the mob controller will bail, and stonedmc master will sigh at poor code and pause the entire controller until your sleep is done.

I cannot merge something that sleeps in Life()

It's best if you come up with a way that doesn't involve sleep, but the alternatives are using spawn, or `set waitfor = 0` which means the proc does above does NOT wait for this proc. Be wary of doing this, because the second you sleep, your proc will return `.` at that exact moment, and there's no potential to return other data (without shenanigans like mutual object references).

Screemonster replied:
The handle_feral proc is called under a spawn. Will that still wreck the mob controller?

If so, handle_hallucinations still needs to be fixed as well as it does the exact same thing and is called directly by life() - again, under a spawn(0).

aztectornado commented:
Damn but this looks cool. Almost makes me wanna go Xenochimera, but I'd have to think of a character to play for that. 

Does the Lazarus thing work on simple mobs too, or is it just a xeno thing?

Screemonster commented:
Got rid of the sleep() and replaced it with a spawn(). Since this code was copypasted from hallucination code, hallucination code could probably do with an overhaul.

While I was in there, I also got rid of the client-view-spinning thing since it broke every sprite on the screen, and replaced it with SCARY VOICES. Like this:

![image](https://user-images.githubusercontent.com/14942479/36418706-8a10a8b8-1627-11e8-882d-bc730b87adbd.png)

Screemonster commented:
The "discount lazarus" reaction only works in human mobs that have the "reconstitute form" verb, and has the same effect as pressing the verb and hitting yes on the confirmation box. It doesn't work on simple mobs.

Arokha commented on code/modules/mob/living/carbon/human/species/station/station_special_abilities_vr.dm:
```diff
@@ -9,119 +9,68 @@
-			C << "You are already reconstructing, or your body is currently recovering from the intense process of your previous reconstitution."
-			return
+/mob/living/carbon/human/proc/chimera_regenerate()
+	var/mob/living/carbon/human/C = src
```
Why cast this? There's no need. If this is defined on human, you can just refer to everything with implicit things. Like one line below, just doing nutrition.

Arokha commented on code/modules/mob/living/carbon/human/species/station/station_special_abilities_vr.dm:
```diff
@@ -9,119 +9,68 @@
-				C << "Your body is too damaged to regenerate without additional nutrients to feed what few living cells remain."
-				return
+	if(C.reviving == 1) //If they're already unable to
+		C << "You are already reconstructing, or your body is currently recovering from the intense process of your previous reconstitution."
```
to_chat(src,"Words"), here and everywhere else

Arokha commented on code/modules/mob/living/carbon/human/species/station/station_special_abilities_vr.dm:
```diff
@@ -9,119 +9,68 @@
+				for(var/datum/wound/W in affected.wounds) // Fix internal bleeds at the start of the rejuv process.
+					if(istype(W, /datum/wound/internal_bleeding))
+						affected.wounds -= W
+						affected.update_damages()
```
I dunno about this. It seems... shady. Those datums may or may not get GC'd, I don't know if they're referenced anywhere else. Maybe consider just healing them?

Screemonster replied:
Any suggestions how? This is how it's done in the equivalent changeling power.

Arokha commented on code/modules/mob/living/carbon/human/species/station/station_special_abilities_vr.dm:
```diff
@@ -9,119 +9,68 @@
-				if(C.stat != DEAD) //If they're still alive after regenning.
+				if(C) //Runtime prevention.
 					C << "<span class='notice'>Consciousness begins to stir as your new body awakens, ready to hatch..</span>"
 					C.verbs += /mob/living/carbon/human/proc/hatch
```
What if they are revived via other means in the meantime?

Screemonster replied:
Now _that's_ an edge-case I don't think either me or cameron accounted for as I can't remember if defibs were around at the time.

Arokha commented on code/modules/mob/living/carbon/human/species/station/station_special_abilities_vr.dm:
```diff
@@ -9,119 +9,68 @@
+			C << "Your body is too damaged to regenerate without additional nutrients to feed what few living cells remain."
+			return
 
+	else if(C.stat != DEAD) //If they're alive at the time of reviving.
```
Misleading comment, as this is not 'reviving' anyone.

Arokha commented on code/modules/mob/living/carbon/human/species/station/station_special_abilities_vr.dm:
```diff
@@ -209,188 +158,189 @@
-						spawn(rand(10,30))
+	if(client && feral >= 10) // largely a copy of handle_hallucinations() without the fake attackers. Unlike hallucinations, only fires once - if they're still feral they'll get hit again anyway.
+		spawn(rand(200,500)/(feral/10))
+			var/halpick = rand(1,100)
```
What if they aren't feral anymore when this procs?

Screemonster replied:
In practical terms? It'll take longer than spawn(500) for them to tick 10 feralness down normally, but can add an "if(!feral) return" easily enough though to be sure.

Arokha commented on code/modules/mob/living/carbon/human/species/station/station_special_abilities_vr.dm:
```diff
@@ -209,188 +158,189 @@
+						var/list/slots_free = list(ui_lhand,ui_rhand)
+						if(l_hand) slots_free -= ui_lhand
+						if(r_hand) slots_free -= ui_rhand
+						if(istype(src,/mob/living/carbon/human))
```
Gross istype of src.

Screemonster replied:
Hallucination code.

Arokha commented on code/modules/mob/living/carbon/human/species/station/station_special_abilities_vr.dm:
```diff
@@ -209,188 +158,189 @@
+							if(!H.r_store) slots_free += ui_storage2
+						if(slots_free.len)
+							halitem.screen_loc = pick(slots_free)
+							halitem.layer = 50
```
Unlikely this will show up on the HUD at all, considering the HUD has a very high plane and you're not setting any on this. You should use hud_layerise()

Screemonster replied:
Hallucination code.

Arokha commented on code/modules/mob/living/carbon/human/species/station/station_special_abilities_vr.dm:
```diff
@@ -209,188 +158,189 @@
+									halimage = image('icons/turf/space.dmi',target,"[rand(1,25)]",TURF_LAYER)
+								if(2)
+									//src << "Fire"
+									halimage = image('icons/effects/fire.dmi',target,"1",TURF_LAYER)
```
The flick_overlay proc can replace most of this if you have suitable animations to go with. Fire would probably work. Not necessary.

Screemonster replied:
Hallucination code.

Arokha commented on code/modules/mob/living/carbon/human/species/station/station_special_abilities_vr.dm:
```diff
@@ -209,188 +158,189 @@
+					if(!halbody)
+						var/list/possible_points = list()
+						for(var/turf/simulated/floor/F in view(src,world.view))
+							possible_points += F
```
This is scary. It'd be a lot cheaper to just pick an x/y randomly within world.view until you found what you want than processing a whole full-screen view most of the time (making sure to bail if you fail too many times (or world.view*world.view times, since that'd be the whole area)).

Screemonster replied:
Hallucination code.

Arokha commented on code/modules/mob/living/carbon/human/species/station/station_special_abilities_vr.dm:
```diff
@@ -209,188 +158,189 @@
+					if(!halbody)
+						var/list/possible_points = list()
+						for(var/turf/simulated/floor/F in view(src,world.view))
+							possible_points += F
```
Same

Screemonster replied:
Hallucination code.

Arokha commented on code/modules/reagents/Chemistry-Recipes_vr.dm:
```diff
@@ -47,6 +47,22 @@
 	holder.clear_reagents()
 
+/datum/chemical_reaction/xenolazarus
+	name = "Discount Lazarus Injector"
```
I'm not sure I'd name a REACTION an INJECTOR.

Screemonster commented:
Moved the actual hatching part into its own proc rather than having redundant parts everywhere. Also fixed an edge case where a dead 'mera that was regenerating gets revived by whatever means - if this happens, the proc just aborts and resets the cooldown when the timer is up. Also put a sanity check in handle_feral just in _case_ they unferal between the proc being called and the sleep taking effect.

Updated a whole bunch of "C <<" into "to_chat" and got rid of redundant var/C = src shenanigans.
